### PR TITLE
feat: per-project dependency management

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: src/commands/dependency/cargo/install.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    const { args, flags } = await this.parse(Install);
+    const isInteractive = getIsInteractive(flags);
+    await ensureFluenceProject(this, isInteractive);
+
+    const packageNameAndVersion: unknown =
+      args[PACKAGE_NAME_AND_VERSION_ARG_NAME];
+
+    const fluenceConfig = await initFluenceConfig(this);
+
+    assert(
```
Comment (coder11) [inline src/commands/dependency/cargo/install.ts:71]: could you add a comment explaining why we are doing this check

File: src/commands/dependency/cargo/install.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      (packageNameAndVersion === undefined ||
+        typeof packageNameAndVersion === "string") &&
+        fluenceConfig !== null
+    );
+
+    if (packageNameAndVersion === undefined) {
```
Comment (coder11) [inline src/commands/dependency/cargo/install.ts:77]: and this one

File: src/commands/dependency/npm/install.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    const { args, flags } = await this.parse(Install);
+    const isInteractive = getIsInteractive(flags);
+    await ensureFluenceProject(this, isInteractive);
+
+    const packageNameAndVersion: unknown =
+      args[PACKAGE_NAME_AND_VERSION_ARG_NAME];
+
+    const fluenceConfig = await initFluenceConfig(this);
+
+    assert(
```
Comment (coder11) [inline src/commands/dependency/npm/install.ts:70]: and these two

**Review by coder11 — APPROVED** [review]

File: src/lib/rust.ts [hunk 4]
```diff
@@ -118,124 +113,129 @@ const isRustInstalled = async (): Promise<boolean> => {
...
-  isNightlyX86?: true;
+  toolchain?: string;
 };
 
-export const cargoDependencies: Record<CargoDependency, CargoDependencyInfo> = {
+export const fluenceCargoDependencies: Record<string, CargoDependencyInfo> = {
   [MARINE_CARGO_DEPENDENCY]: {
     recommendedVersion: MARINE_RECOMMENDED_VERSION,
-    packageName: MARINE_CARGO_DEPENDENCY,
-    isNightlyX86: true,
```
[the remark this answers is not in the document]
Comment (shamsartem) [reply src/lib/rust.ts:163]: Not removing but replacing it with a toolchain flag, which is more flexible. People would be able to install other dependencies with different toolchaing if they want to

**Pull request merged**